### ISSUE TITLE
Lint `bare_trait_objects` doesn't warn on traits used as types in paths

### ISSUE THREAD

**Issue by petrochenkov** [rust-lang/rust#65371]
Example:
```rust
trait Method { fn method() {} }

trait Tr {}

impl Method for Tr {} // warning: trait objects without an explicit `dyn` are deprecated

fn main() {
    Tr::method; // No warning
}
```
According to the logic of the lint there should be a suggestion to rewrite `Tr::method` as `<dyn Tr>::method`.

**Comment by petrochenkov** [comment]
Fixed in https://github.com/rust-lang/rust/pull/82868.
